src/stats.hpp

```diff
@@ -28,45 +28,45 @@ class Config;
 class Context;
 
 // Statistics fields in storage order.
-enum stats {
-  STATS_NONE = 0,
-  STATS_STDOUT = 1,
-  STATS_STATUS = 2,
-  STATS_ERROR = 3,
-  STATS_CACHEMISS = 4,
-  STATS_PREPROCESSOR = 5,
-  STATS_COMPILER = 6,
-  STATS_MISSING = 7,
-  STATS_CACHEHIT_CPP = 8,
-  STATS_ARGS = 9,
-  STATS_LINK = 10,
-  STATS_NUMFILES = 11,
-  STATS_TOTALSIZE = 12,
-  STATS_OBSOLETE_MAXFILES = 13,
-  STATS_OBSOLETE_MAXSIZE = 14,
-  STATS_SOURCELANG = 15,
-  STATS_BADOUTPUTFILE = 16,
-  STATS_NOINPUT = 17,
-  STATS_MULTIPLE = 18,
-  STATS_CONFTEST = 19,
-  STATS_UNSUPPORTED_OPTION = 20,
-  STATS_OUTSTDOUT = 21,
-  STATS_CACHEHIT_DIR = 22,
-  STATS_NOOUTPUT = 23,
-  STATS_EMPTYOUTPUT = 24,
-  STATS_BADEXTRAFILE = 25,
-  STATS_COMPCHECK = 26,
-  STATS_CANTUSEPCH = 27,
-  STATS_PREPROCESSING = 28,
-  STATS_NUMCLEANUPS = 29,
-  STATS_UNSUPPORTED_DIRECTIVE = 30,
-  STATS_ZEROTIMESTAMP = 31,
-  STATS_CANTUSEMODULES = 32,
+enum class Stats {
+  NONE = 0,
```

Comment:
I didn't want to change too much. But i can certainly rename the values.

Reply:
Now that I look at the values... the 0 value is "none". So why bother making it an optional in several places? Just return none.

Reply:
Instead of just making the names `lower_snake_case` I have now reused the `id` string. In general that name seems way better understandable. And it's probably easier to trace such stuff if the names match.

Reply:
> Now that I look at the values... the 0 value is "none". So why bother making it an optional in several places? Just return none.

That would work as well, it's just a matter of taste.

> Instead of just making the names `lower_snake_case` I have now reused the id string.

Good idea.